[MMB-219] Update docs (#113)
ably-labs/spaces: docs/usage.md

## Members

> Members is a core concept of the library. When you enter a space, you become a `member`. On the client, your own membership is to referred to as `self`. You can get your `self` by calling `space.getSelf`. To get all the members (including self), call `space.getMembers`. These method will return (respectively an object and array of):

Reviewer: Similar to other files, we need some more information on the `space` object before moving to `space.members`

Emitted when a member leaves a space. Called with the member leaving the space.

#### update

> Emitted when a member profile updates a space. Called with the member updating the space.

Reviewer: ```suggestion
Emitted when members enter, leave and their location is updated. Called with the member updating the space.
```

> #### all

Reviewer: Do we have an event called `all`? 🤔

});
```

## Members

> Members is a core concept of the library. When you enter a space, you become a `member`. On the client, your own membership is to referred to as `self`. You can get your `self` by calling `space.members.getSelf()`. To get all the members (including self), call `space.members.getAll()`. These method will return (respectively an object and array of):

Reviewer: ```suggestion
When you enter a space, you become a `member`. On the client, your own membership is to referred to as `self`. You can get your `self` by calling `space.members.getSelf()`. To get all the members (including self), call `space.members.getAll()`. These method will return (respectively an object and array of):
```

});
```

## Members

> When you enter a space, you become a `member`. On the client, your own membership is to referred to as `self`. You can get your `self` by calling `space.members.getSelf()`. To get all the members (including self), call `space.members.getAll()`. These method will return (respectively an object and array of):

Reviewer 2: ```suggestion
When you enter a space, you become a `member`. On the client, your own membership is to referred to as `self`. You can get your `self` by calling `space.members.getSelf()`. To get all the members (including self), call `space.members.getAll()`. These methods will return (respectively an object and array of):
```

The location property will be set on the [member](#members).

> Because locations are part of members, an `update` member event will be emitted when a member updates their location. When a member leaves, their location is set to `null`.

Reviewer: `update` is an optional event name. The subscription will also be triggered with no event name. So we can just say "Because locations are part of members, a member event will be emitted"

Reviewer: I don't think this is true anymore - this is applicable to `space`, not `space.members`

```ts

> space.members.subscribe('update', (memberProfileUpdate) => {

Reviewer: It's not just a profile update. This will get triggered for enters, leaves, removes also.

The location property will be set on the [member](#members).

> Because locations are part of members, a member event will be emitted when a member updates their location. When a member leaves, their location is set to `null`.

Reviewer 3: I'm not sure this is true any longer. Location is part of a member, however "a member event" makes it sound like it would be picked up by `members.subscribe()`. AFAIK it won't be, it would only be picked up by `space.subscribe()`.

> Because locations are part of members, a member event will be emitted when a member updates their location. When a member leaves, their location is set to `null`.

Reviewer: ```suggestion
A location event will be emitted when a member updates their location. When a member leaves, their location is set to `null`.
```

```ts
space.subscribe('update', (memberProfileUpdate) => {

>   console.log(memberProfileUpdate);

Reviewer 3: `memberProfileUpdate` looks odd in this example as that's essentially how we refer to arbitrary profile data. Maybe `member` and `member.location`?

});
```

> However, it's possible to listen to just location updates. `locations` is an [event emitter](#event-emitters) and will emit the `locationUpdate` event:

Reviewer: We need to get rid of the example above and keep only this one with the correct description.

Calling `subscribe` with a single function argument will subscribe to all events on that emitter.

```ts

> space.subscribe(() => {});

Reviewer: There's also `space.subscribe()`
